pkg/provider/azure/keyvault/keyvault.go

```diff
@@ -170,34 +171,53 @@ func (a *Azure) GetSecretMap(ctx context.Context, ref esv1alpha1.ExternalSecretD
 
 func (a *Azure) newAzureClient(ctx context.Context) (*keyvault.BaseClient, string, error) {
 	spec := *a.store.GetSpec().Provider.AzureKV
-	tenantID := *spec.TenantID
 	vaultURL := *spec.VaultURL
 
-	if spec.AuthSecretRef == nil {
-		return nil, "", fmt.Errorf("missing clientID/clientSecret in store config")
-	}
-	clusterScoped := false
-	if a.store.GetObjectKind().GroupVersionKind().Kind == esv1alpha1.ClusterSecretStoreKind {
-		clusterScoped = true
-	}
-	if spec.AuthSecretRef.ClientID == nil || spec.AuthSecretRef.ClientSecret == nil {
-		return nil, "", fmt.Errorf("missing accessKeyID/secretAccessKey in store config")
-	}
-	cid, err := a.secretKeyRef(ctx, a.store.GetNamespace(), *spec.AuthSecretRef.ClientID, clusterScoped)
-	if err != nil {
-		return nil, "", err
-	}
-	csec, err := a.secretKeyRef(ctx, a.store.GetNamespace(), *spec.AuthSecretRef.ClientSecret, clusterScoped)
-	if err != nil {
-		return nil, "", err
-	}
-
-	clientCredentialsConfig := kvauth.NewClientCredentialsConfig(cid, csec, tenantID)
 	// the default resource api is the management URL and not the vault URL which we need for keyvault operations
-	clientCredentialsConfig.Resource = "https://vault.azure.net"
-	authorizer, err := clientCredentialsConfig.Authorizer()
-	if err != nil {
-		return nil, "", err
+	resource := "https://vault.azure.net"
+
+	var authorizer kvautorest.Authorizer = nil
+	if spec.AuthType ==  esv1alpha1.ManagedIdentity {
```

Comment:
Can you please restructure the if / else block when handling authentication strategies? We did this in a couple of other providers (e.g. [vault](https://github.com/external-secrets/external-secrets/blob/main/pkg/provider/vault/vault.go#L321))
that'll also make sonarcloud/codesmells happy ;)

Reply:
Pushed a fix which looks like the linked solution. SonarCloud is happy now :-)